vectorizer.py

```diff
@@ -52,8 +55,10 @@ def __init__(self, model_path: str, cuda_support: bool, cuda_core: str, cuda_per
 
         self.tokenizer = self.model_delegate.create_tokenizer(model_path)
 
+        self.executor = ThreadPoolExecutor(max_workers=1)
```

Comment:
1. Previously the vectorization process was synchronous, so in theory vectors were generated only one at a time. I left it at 1 to mimic this; a side effect of this is that the GPU memory usage when cuda is enabled is more predictable. I'll run some experiments with other values though, it could lead to significant speedups!

2. Good shout, I tend to use ThreadPoolExecutor over ProcessPoolExecutor since it's technically less overhead (and since with 1 worker the difference is negligible) but I'll try both for the many workers case.

Reply:
Trying ProcessPoolExecutor I see a couple issues:

```
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/alex/repos/forked-t2v-transformers-models/app.py", line 66, in read_item
    vector = await vec.vectorize(item.text, item.config)
  File "/home/alex/repos/forked-t2v-transformers-models/vectorizer.py", line 108, in vectorize
    return await asyncio.wrap_future(self.executor.submit(self._vectorize, text, config))
  File "/usr/lib/python3.10/multiprocessing/queues.py", line 244, in _feed
    obj = _ForkingPickler.dumps(obj)
  File "/usr/lib/python3.10/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
  File "/home/alex/repos/t2v-transformers-models/env/lib/python3.10/site-packages/torch/multiprocessing/reductions.py", line 261, in reduce_tensor
  File "/home/alex/repos/t2v-transformers-models/env/lib/python3.10/site-packages/torch/storage.py", line 920, in _share_cuda_
RuntimeError: unable to open shared memory object </torch_49047_1633709960_558> in read-write mode: Too many open files (24)
INFO:     127.0.0.1:42060 - "POST /vectors HTTP/1.1" 500 Internal Server Error
```
and
```
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/alex/repos/forked-t2v-transformers-models/app.py", line 66, in read_item
    vector = await vec.vectorize(item.text, item.config)
  File "/home/alex/repos/forked-t2v-transformers-models/vectorizer.py", line 108, in vectorize
    return await asyncio.wrap_future(self.executor.submit(self._vectorize, text, config))
  File "/usr/lib/python3.10/multiprocessing/queues.py", line 244, in _feed
    obj = _ForkingPickler.dumps(obj)
  File "/usr/lib/python3.10/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
TypeError: cannot pickle '_thread.lock' object
```

However, ThreadPoolExecutor handles multiple workers quite nicely.

Reply:
Done, I removed the kwarg altogether since the default is None.